### PR TITLE
feat: xlsx-to-oscal-profile

### PR DESCRIPTION
**Pull request by degenaro**
- create common xlsx module
- create xlsx-to-oscal-profile task
- artifacts for 100% code coverage

## Types of changes

- [ ] Hot fix (emergency fix and release)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation (change which affects the documentation site)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Release (`develop` -> `main`)

## Quality assurance (all should be covered).

- [x] My code follows the code style of this project.
- [x] Documentation for my change is up to date?
- [x] My PR meets testing requirements.
- [x] All new and existing tests passed.
- [x] All commits are signed-off.

## Summary

New task xlsx-to-oscal-profile with refactoring of common spread sheet code from task xlsx-to-component-definition.

## Key links:

- [Sonar coverage](https://sonarcloud.io/dashboard?id=compliance-trestle)

# Before you merge

- Ensure it is a 'squash commit' if not a release.
- Ensure CI is currently passing
- Check sonar. If you are working for a fork a maintainer will reach out, if required.

### CODE REVIEW

File: trestle/tasks/xlsx_to_oscal_profile.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+            logger.info(f'output: {ofile}')
+        profile.oscal_write(pathlib.Path(ofile))
+        # issues
+        self._report_issues()
+        return TaskOutcome('success')
+
+    def _get_include_controls(self) -> List[SelectControlById]:
+        """Get include controls from spread sheet."""
+        include_control = SelectControlById(with_ids=self._get_with_ids())
+        return [include_control]
```
Comment (fsuits) [inline trestle/tasks/xlsx_to_oscal_profile.py:134]: Do you want to specify a with_child_controls value here?

**Review by fsuits — APPROVED** [review]
LGTM with one small question.

File: trestle/tasks/xlsx_to_oscal_profile.py [hunk 2]
```diff
@@ -0,0 +1,180 @@
+# -*- mode:python; coding:utf-8 -*-
+# Copyright (c) 2021 IBM Corp. All rights reserved.
```
Comment (enikonovad) [inline trestle/tasks/xlsx_to_oscal_profile.py:2]: 2022

**Review by enikonovad — APPROVED** [review]
Minor comments

**Comment by enikonovad** [comment]
@degenaro Can we add a brief documentation for the existing tasks as part of this PR? I believe the documentation should at least include the available tasks names ie 'xlsx-to-oscal-cd' and new 'xlsx-to-oscal-profile' and explain what config is available for them

File: trestle/tasks/xlsx_to_oscal_profile.py (hunk 1 above)
Comment (degenaro) [reply trestle/tasks/xlsx_to_oscal_profile.py:134]: For now, no. It is not required.

File: trestle/tasks/xlsx_to_oscal_profile.py (hunk 2 above)
Comment (degenaro) [reply trestle/tasks/xlsx_to_oscal_profile.py]: Fixed.

File: trestle/tasks/xlsx_helper.py [hunk 3]
```diff
@@ -0,0 +1,407 @@
...
+from openpyxl.cell.cell import MergedCell
+from openpyxl.utils import get_column_letter
+
+from trestle.common.catalog_helper import CatalogHelper
+from trestle.tasks.base_task import TaskBase
+
+logger = logging.getLogger(__name__)
+
+
+def is_sublist(needle: List[str], haystack: List[str]) -> bool:
```
[the remark this answers is not in the document]
Comment (degenaro) [reply trestle/tasks/xlsx_helper.py]: Fixed.

**Comment by degenaro** [comment]
> @degenaro as we are moving close to the 1.0 release, can we add a brief documentation for the existing tasks as part of this PR? I believe the documentation should at least include the available tasks names ie `xlsx-to-oscal-cd` and new `xlsx-to-oscal-profile` and explain what config is available for them

Actually, there already is cli.md doc for the comp-def transformer. But is it wrong due to the new name, so I will fix. I'll also add doc for the profile transformer...

**Pull request merged**